encryption/ecies/publickey.go

```diff
@@ -57,6 +65,47 @@ func PublicKeyFromHex(hexKey string, curve elliptic.Curve) (*PublicKey, error) {
 	return PublicKeyFromBytes(b, curve)
 }
 
+// PublicKeyFromPEMFile parses a public key from a PEM file
+func PublicKeyFromPEMFile(path string) (*PublicKey, error) {
+	pemBytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return PublicKeyFromPEMBytes(pemBytes)
+}
+
+// PublicKeyFromPEMString parses a public key from a PEM string
+func PublicKeyFromPEMString(pemString string) (*PublicKey, error) {
+	return PublicKeyFromPEMBytes([]byte(pemString))
+}
+
+// PublicKeyFromPEMBytes parses a public key from its PEM form
+func PublicKeyFromPEMBytes(bytes []byte) (*PublicKey, error) {
+	block, _ := pem.Decode(bytes)
+	if block == nil {
+		return nil, errors.New("failed to parse PEM block containing the key")
+	}
+
+	pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("failed to parse DER encoded public key: %s", hex.EncodeToString(bytes)), err)
+	}
+
+	switch pub.(type) {
+	case *ecdsa.PublicKey:
```

Comment:
I checked `crypto/` library code and it seems `x509.ParsePKIXPublicKey` only returns PublicKey, but weird they put `Any` type.

I think we can use `if` instead of `switch`?

Reply:
ohh same name from different packages